nucypher/characters/lawful.py

```diff
@@ -607,9 +607,9 @@ def gather_decryption_shares(
             gathered_shares.append(decryption_share)
             self.log.debug(f"Got {len(gathered_shares)}/{threshold} shares so far...")
 
-            # TODO: Uncomment these lines to reproduce the bug
-            # if variant == FerveoVariant.SIMPLE and (len(gathered_shares) == threshold):
-            #     break
+            # FIXME: BUG HAPPENS HERE
+            if variant == FerveoVariant.SIMPLE and (len(gathered_shares) == threshold):
+                break
```

Comment:
@cygnusv @piotr-roslaniec The bug has been isolated to these lines (with ferveo==0.1.8)